### ISSUE TITLE
Send DAP OutputEvents on stdout/stderr of debugged process when console is `internalConsole`

### ISSUE THREAD

**Issue by gfszr** [go-delve/delve#3094]
**Go Version**: 1.18
**Delve Version**: 1.8.2

#### Current status
DAP allows sending `stdout`/`stderr` outputs back to the DAP client, by which it is shown on the debug console of the IDE. Delve spawns the debuggees with its own stdout/stderr, so the output is missing from the normal debug console. This is the desired behavior in case the terminal being used is either `integratedTerminal` or `externalTerminal`, but in the case of `internalConsole`, in order to allow showing the output to the user, clients required to somehow still redirect output (For example, [nvim-dap-go](https://github.com/leoluz/nvim-dap-go/blob/main/lua/dap-go.lua#L57)).

#### Desired status
Delve would send [OutputEvent](https://microsoft.github.io/debug-adapter-protocol/specification#Events_Output)s of data from the debuggee's `stdout`/`stderr` if the console given is `internalConsole`, which would allow the DAP client to easily present it.

**Comment by aarzilli** [comment]
The problem with this is that some program will behave differently if their stdout/stderr is not a tty. It is unfortunate that DAP was designed like this.

**Comment by gfszr** [comment]
@aarzilli Yet, other debug adapters do the same, even if running with a tty - debugpy for instance.

**Comment by gfszr** [comment]
The specific case in [debugpy](https://github.com/microsoft/debugpy) can be seen [here](https://github.com/microsoft/debugpy/blob/main/src/debugpy/adapter/clients.py#L221):
```python
 redirecting = arguments.get("console") == "internalConsole"
```
Redirection is decided in case the user specifies `internalConsole` as its terminal - by which this means that the user requested output to be redirected in the console. I agree this shouldn't be default behavior unless the user requested `internalConsole`, so perhaps updating this issue with "redirect output when console defined as internalConsole` suffices? @aarzilli

**Comment by gfszr** [comment]
I revised the issue to only send such `OutputEvent`'s when launch configuration is `internalConsole`.

**Comment by aarzilli** [comment]
This was implemented by #3253
